use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,358 @@
+
+##Helpful Links before starting
+[https://github.com/intel/QAT_Engine](https://github.com/intel/QAT_Engine)
+
+[https://01.org/intel-quickassist-technology](https://01.org/intel-quickassist-technology)
+
+[https://github.com/intel/intel-device-plugins-for-kubernetes/](https://github.com/intel/intel-device-plugins-for-kubernetes/)
+
+[https://dpdk-docs.readthedocs.io/en/latest/cryptodevs/qat.html](https://dpdk-docs.readthedocs.io/en/latest/cryptodevs/qat.html)
+
+
+##Steps to enable Intel QAT in Kata Containers
+Some of the steps only have to be done once. Some have to be done every reboot. Other steps are required only when the host kernel changes.
+
+
+1. [Script Variables](#script-variables)
+    1. [Set environment variables (Every Reboot)](#script-variables-1)
+1. [Prepare the Host (Clear Linux)](#prepare-host)
+    1. [Identify which PCI bus the Intel QAT card is on](#prepare-1)
+    1. [Replace Clear Linux kernel with LTS kernel](#prepare-2a)
+    1. [Install necessary bundles for Clear Linux and disable kernel updates](#prepare-2)
+    1. [Allow insecure modules to load to host](#prepare-3)
+    1. [Download Intel QAT drivers](#prepare-4)
+    1. [Compile Intel QAT modules against host kernel](#prepare-5)
+    1. [Copy Intel QAT configuration files and enable vf’s](#prepare-6)
+    1. [Expose and Bind Intel QAT virtual functions to vfio-pci (Every reboot)](#prepare-7)
+    1. [Check Intel QAT virtual functions are enabled](#prepare-8)
+1. [Prepare Kata Containers](#prepare-kata)
+    1. [Download Kata Kernel Source](#prepare-kata-1)
+    1. [Build Kata Kernel](#prepare-kata-1a)
+    1. [Copy Kata Kernel](#prepare-kata-2)
+    1. [Prepare Kata root filesystem](#prepare-kata-3)
+    1. [Compile Intel QAT drivers for Kata Kernel and add to Kata rootfs](#prepare-kata-4)
+    1. [Copy Kata rootfs](#prepare-kata-4a)
+    1. [Update Kata configuration.toml to point to custom kernel and rootfs](#prepare-kata-5)
+1. [Verify Intel QAT works in a Docker\* Kata container](#verify-qat)
+    1. [Build OpenSSL Intel QAT engine container](#verify-qat-1)
+    1. [Test Intel QAT in Docker](#verify-qat-2)
+    1. [Troubleshooting](#verify-qat-3)
+1. [Optional Scripts](#optional-scripts)
+    1. [Verify Intel QAT card counters are incremented](#optional-scripts-1)
+    1. [Unbind all vf’s from vfio driver, reset, and rebind to the host](#optional-scripts-2)
+
+##Script variables<a name="script-variables"></a>
+The variables listed below must be set before running through the scripts. These variables refer to locations to store modules and configuration files on the host and as well as links to the drivers to use. Modify these as needed to point to updated drivers or different install locations.
+
+###Set environment variables (Every Reboot)<a name="script-variables-1"></a>
+```
```

Comment:
Please can you use a shell codeblock (` ```sh `) for all shell code.

Reply:
I did for this one but it is not clear to me if I should in other parts.

Reply:
We shell blocks for all shell code (since it renders in colour in GitHub). Bash blocks we tend to reserve for code blocks that are actually executed by our CI - see:

- https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh
- https://github.com/kata-containers/documentation/blob/master/install/installing-with-kata-doc-to-script.md